### PR TITLE
Implement Search Users / Instance Users

### PR DESCRIPTION
**Pull request by allenwq**

### CODE REVIEW

File: app/models/instance_user.rb [hunk 1]
```diff
@@ -1,8 +1,13 @@
 class InstanceUser < ActiveRecord::Base
+  include InstanceUserSearchConcern
   acts_as_tenant :instance, inverse_of: :instance_users
 
   enum role: { normal: 0, instructor: 1, administrator: 2 }
   belongs_to :user, inverse_of: :instance_users
 
   scope :ordered_by_username, -> { joins(:user).merge(User.order(name: :asc)) }
+
+  def self.search_and_ordered_by_username(keyword)
```
Comment (allenwq) [inline app/models/instance_user.rb]: @lowjoel Have a small issue here while combining `.search` and `.ordered_by_username`, the field in `order_by` must appear in `group_by`, so I added this method here to resolve it.

Comment (lowjoel) [inline app/models/instance_user.rb]: ...I don't see why that's necessary

Comment (allenwq) [inline app/models/instance_user.rb]: See the exception:
```
$ InstanceUser.group { id }.ordered_by_username
ActiveRecord::StatementInvalid: PG::GroupingError: ERROR:  column "users.name" must appear in the GROUP BY clause or be used in an aggregate function
```

Comment (lowjoel) [inline app/models/instance_user.rb]: and the query?

Comment (allenwq) [inline app/models/instance_user.rb]:
```
SELECT "instance_users".* FROM "instance_users" INNER JOIN "users" ON "users"."id" = "instance_users"."user_id" WHERE "instance_users"."instance_id" = $1 GROUP BY "instance_users"."id"  ORDER BY "users"."name" ASC
```

Comment (lowjoel) [inline app/models/instance_user.rb]: Can't fix it myself, okay then.

**Pull request merged**